source/kernel/CgiExecution.cpp

```
@@ -131,25 +133,28 @@ char** CgiExecution::_get_env(webshell::Request& request)
         env = new char*[envp.size() + 1];
         if (env == NULL)
```

`new` itself will throw `bad_alloc` instead of returning NULL C-style. @yenthing therefore you do not need to protect this. I believe bad_alloc we handle centralized. We did not end up discussing a strategy for this though, or maybe i just forgot @LeaYeh